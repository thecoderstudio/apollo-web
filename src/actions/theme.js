function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
