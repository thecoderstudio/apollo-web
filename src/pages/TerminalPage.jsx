import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Terminal from '../components/terminal/Terminal';

const propTypes = {
  agentEndpoint: PropTypes.string,
  readOnly: PropTypes.bool
};

class TerminalPage extends React.PureComponent {
  constructor(props) {
    super(props);
    this.terminalRef = React.createRef();
  }

  componentDidMount() {
    this.terminalRef.current.fit();
  }

  render() {
    const { match: { params } } = this.props;
    return <Terminal
      ref={this.terminalRef}
      readOnly={this.props.readOnly}
      agentEndpoint={this.props.agentEndpoint}
      agent={this.props.agents.get(params.agentId)} />;
  }
}

TerminalPage.propTypes = propTypes;

export default connect(
  state => ({agents: state.agent})
)(TerminalPage);
