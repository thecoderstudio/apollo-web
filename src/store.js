import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2'
import storage from 'redux-persist/lib/storage';
import mapTransformer from './transformers/map';
import rootReducer from './reducers/root';

const persistConfig = {
  key: 'root',
  whitelist: ['auth', 'agent'],
  transforms: [mapTransformer({ whitelist: 'agent' })],
  storage
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
export const store = createStore(persistedReducer);
export const persistor = persistStore(store);
