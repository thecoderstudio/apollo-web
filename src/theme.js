export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
