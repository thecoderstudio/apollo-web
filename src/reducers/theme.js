import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
