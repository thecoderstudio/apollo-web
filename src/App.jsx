import React from 'react';
import { connect } from 'react-redux';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import styled, { createGlobalStyle, ThemeProvider } from 'styled-components';
import ProtectedRoute from './components/ProtectedRoute';
import NavBar from './components/NavBar';
import Notifications from './components/notifications/Notifications';
import Dashboard from './pages/Dashboard';
import Welcome from './pages/Welcome';
import AgentDetail from './pages/AgentDetail';
import TerminalPage from './pages/TerminalPage';
import Admin from './pages/Admin';
import NotFound from './pages/NotFound';
import { darkTheme } from './theme';

const GlobalStyle = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css2?family=B612:wght@400;700&family=Roboto:wght@300;400&display=swap');

  html, body {
    margin: 0px;
    padding: 0px;
    height: 100%;
    width: 100%;
    font-weight: 500;
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.black};
  }

  body {
    font-family: 'Roboto', sans-serif;
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: 'B612', sans-serif;
  }
`;

const Content = styled.div`
  height: 100%;
  width: 100%;
`;

function App(props) {
  return (
    <div>
      <ThemeProvider theme={darkTheme}>
        <BrowserRouter>
          <Content>
            {props.authenticated && props.hasChangedPassword && <NavBar />}
            <Switch>
              <ProtectedRoute exact path="/" component={Dashboard} fallbackComponent={Welcome} />
              <ProtectedRoute exact path="/agent/:agentId/shell" component={TerminalPage} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/agent/:agentId" component={AgentDetail} fallbackComponent={NotFound} />
              <ProtectedRoute exact path="/admin" component={Admin} fallbackComponent={NotFound} role="admin" />
              <Route component={NotFound} />
            </Switch>
            <Notifications />
            <GlobalStyle />
          </Content>
        </BrowserRouter>
      </ThemeProvider>
    </div>
  );
}

export default connect(
  (state) => ({
    authenticated: state.authenticated,
    hasChangedPassword: state.currentUser.hasChangedInitialPassword
  })
)(App);
