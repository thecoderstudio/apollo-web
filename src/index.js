import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
