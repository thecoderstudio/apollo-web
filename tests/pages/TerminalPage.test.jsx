import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import WS from 'jest-websocket-mock';
import { mount } from 'enzyme';
import TerminalPage from '../../src/pages/TerminalPage';
import { darkTheme } from '../../src/theme';

const mockStore = configureStore([]);

function getComponentTags(store) {
  return (
    <div>
      <Provider store={store}>
        <ThemeProvider theme={darkTheme}>
          <TerminalPage
            match={{
              params: {
                agentId: "fakeid"
              }
            }}
          />
        </ThemeProvider>
      </Provider>
    </div>
  );
}

describe('agent list item', () => {
  const server = new WS(`ws://localhost:1234/agent/a2346886-83ba-442d-9fb7-d024c6274e22/shell`);

  afterEach(() => {
    WS.clean();
  });

  it('correctly renders', () => {
    const agents = new Map();
    agents.set('fakeid', {
      id: 'fakeid',
      name: 'fake',
      connection_state: 'connected'
    });
    const store = mockStore({
      agent: agents
    });
    const wrapper = mount(getComponentTags(store));
    expect(wrapper).toMatchSnapshot();
  });
});
