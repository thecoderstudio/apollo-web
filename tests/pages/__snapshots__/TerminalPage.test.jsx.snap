// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`terminal page correctly renders 1`] = `
.c0 {
  height: 100%;
  width: 100%;
  padding: 20px;
}

.c1 {
  height: 100%;
  width: 100%;
}

.c1 .xterm-viewport {
  overflow: hidden !important;
}

<div>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Component
      theme={
        Object {
          "accent": "#4D02EE",
          "black": "#181818",
          "boxShadow": "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
          "connectedColor": "#27AE60",
          "connectingColor": "#F39C12",
          "darkGrey": "#3E3E3E",
          "darkWhite": "#F1F1F1",
          "disconnectedColor": "#C0392B",
          "error": "#ED3E2A",
          "green": "#32E67E",
          "inactive": "#363636",
          "lightBlack": "#1B1B1B",
          "name": "dark",
          "overlay": "rgba(37, 37, 37, 0.6)",
          "primary": "#02A3EE",
          "red": "#ED3E2A",
          "selectedBlack": "#252525",
          "smallTextSize": "0.85rem",
          "warning": "#ED852A",
          "white": "#F8F8F8",
        }
      }
    >
      <Connect(TerminalPage)
        match={
          Object {
            "params": Object {
              "agentId": "fakeid",
            },
          }
        }
      >
        <TerminalPage
          agents={
            Map {
              "fakeid" => Object {
                "connectionState": "connected",
                "id": "fakeid",
                "name": "fake",
              },
            }
          }
          dispatch={[Function]}
          match={
            Object {
              "params": Object {
                "agentId": "fakeid",
              },
            }
          }
        >
          <WithTheme(Terminal)
            agent={
              Object {
                "connectionState": "connected",
                "id": "fakeid",
                "name": "fake",
              }
            }
          >
            <Terminal
              agent={
                Object {
                  "connectionState": "connected",
                  "id": "fakeid",
                  "name": "fake",
                }
              }
              agentEndpoint="shell"
              onSocketClose={[Function]}
              readOnly={false}
              theme={
                Object {
                  "accent": "#4D02EE",
                  "black": "#181818",
                  "boxShadow": "0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05)",
                  "connectedColor": "#27AE60",
                  "connectingColor": "#F39C12",
                  "darkGrey": "#3E3E3E",
                  "darkWhite": "#F1F1F1",
                  "disconnectedColor": "#C0392B",
                  "error": "#ED3E2A",
                  "green": "#32E67E",
                  "inactive": "#363636",
                  "lightBlack": "#1B1B1B",
                  "name": "dark",
                  "overlay": "rgba(37, 37, 37, 0.6)",
                  "primary": "#02A3EE",
                  "red": "#ED3E2A",
                  "selectedBlack": "#252525",
                  "smallTextSize": "0.85rem",
                  "warning": "#ED852A",
                  "white": "#F8F8F8",
                }
              }
            >
              <styled.div>
                <div
                  className="c0"
                >
                  <styled.div
                    data-testid="terminal"
                  >
                    <div
                      className="c1"
                      data-testid="terminal"
                    />
                  </styled.div>
                </div>
              </styled.div>
            </Terminal>
          </WithTheme(Terminal)>
        </TerminalPage>
      </Connect(TerminalPage)>
    </Component>
  </Provider>
</div>
`;
